specification/columns/commitmentdiscountquantity.md

```diff
@@ -0,0 +1,64 @@
+
+```sql
+SELECT (
+    SELECT SUM(CommitmentDiscountQuantity) FROM <dataset> WHERE ChargeCategory = 'Usage'
+    AND CommitmentDiscountStatus = 'Used' AND CommitmentDiscountId = '<commitment-discount-id>'
+    /
+    SELECT SUM(CommitmentDiscountQuantity) FROM <dataset> WHERE ChargeCategory = 'Usage'
+    AND CommitmentDiscountId = '<commitment-discount-id>'
+)
+```
+
+With this information, customers can analyze their existing *commitment discount* strategy to make more informed decisions on additional purchases.
+
+**Important:** When aggregating Commitment Discount Quantity for utilization calculations, it's important to exclude either one-time or recurring *commitment discount* purchases (i.e. when *ChargeCategory* is *Purchase*) that are paid to cover future eligible charges (e.g., *Commitment Discount*). This exclusion helps prevent double counting of these charges in the aggregation.
+
+The CommitmentDiscountQuantity column adheres to the following requirements:
+
+* CommitmentDiscountQuantity MUST be present in the billing data when the provider supports *commitment discounts*.
+* CommitmentDiscountQuantity MUST be of type Decimal and MUST conform to [Numeric Format](#numericformat) requirements.
+* CommitmentDiscountQuantity MAY be negative if [*ChargeClass*](#chargeclass) is "Correction".
+
+In cases where the *ChargeCategory* is "Purchase" and [*CommitmentDiscountId*](#commitmentdiscountid) is not null, the following applies:
+
+* When [*ChargeFrequency*](#chargefrequency) is "One-Time", and *CommitmentDiscountId* is not null, CommitmentDiscountQuantity MUST be the positive quantity of *CommitmentDiscountUnits*, paid fully or partially upfront, that is eligible for consumption over the *commitment discount's* *term*.
+* When *ChargeFrequency* is "Recurring", and *CommitmentDiscountId* is not null, CommitmentDiscountQuantity MUST be the positive quantity of *CommitmentDiscountUnits* that is eligible for consumption for each *charge period* that corresponds with the purchase.
+
+In cases where the *ChargeCategory* is "Usage" and *CommitmentDiscountId* is not null, the following applies:
+
+* When *CommitmentDiscountStatus* is "Used", and *ChargeClass* is not "Correction", CommitmentDiscountQuantity MUST be the positive, metered quantity of *CommitmentDiscountUnits* that is consumed over the *row's* *charge period*.
+* When *CommitmentDiscountStatus* is "Unused", and *ChargeClass* is not "Correction", CommitmentDiscountQuantity MUST be the remaining, positive, unused quantity of *CommitmentDiscountUnits* for the *row's* *charge period*.
```

Comment:
Is there an example showing how this would look if you had a partially used commitment. These two statements may cause an issue in showing the data correctly, if you had partial used and partial unused where the statement refers to the charge periods row or would there be two distinct rows?

Reply:
Correct, there would be 1 row for used and 1 for unused.  This is the purpose of this [appendix](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/pull/528) which simulates these exact scenarios.